src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -40,14 +49,22 @@ export class NotificationPopupComponent implements OnInit {
      * @param notification {Notification}
      */
     navigateToTarget(notification: Notification): void {
-        this.router.navigateByUrl(this.notificationTargetRoute(notification));
+        // tslint:disable-next-line:triple-equals
+        if (notification.title == LIVE_EXAM_EXERCISE_UPDATE_NOTIFICATION_TITLE) {
+            const target = JSON.parse(notification.target!);
```

Comment:
Do you really have to manually parse the target here? I think you should be able to use typescript's typing mechanisms and type-cast the value...

Reply:
Could you please explain or give an example of how this might work?
I just used the same logic the old code uses to access the target property of notification.
![image](https://user-images.githubusercontent.com/65814168/124451131-603e0c00-dd85-11eb-8637-d01d8cc2ab7d.png)

If I try e.g. `const target = notification.target! as string` I get issues accessing the properties of `target` (e.g. `target.mainPage`)
![image](https://user-images.githubusercontent.com/65814168/124451409-a4c9a780-dd85-11eb-9403-f19d3ee18491.png)

Reply:
I guess in this case, it's ok then.